app/models/volunteer.rb

```diff
@@ -124,66 +129,77 @@ def has_main_region?
   end
 
   def main_region
-    self.regions[0]
+    regions[0]
   end
 
   def region_ids
-    self.regions.collect{ |r| r.id }
+    regions.collect(&:id)
```

Comment:
I think this method can be scrapped, as Rails provides `association_ids` :D 

http://apidock.com/rails/v3.2.13/ActiveRecord/Associations/ClassMethods/has_many